src/map/map.js

```diff
@@ -33,21 +33,40 @@ export default class Map extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      currentChatRoomId: '',
-      lastPosition: {},
-      chatRoomExists: true,
+      coords: '',
+      newRoom: true,
     };
   }
 
   componentDidMount() {
-    this.watchID = navigator.geolocation.watchPosition((position) => {
+    this.listenForGeoChange();
+  }
+
+  componentWillUnmount() {
+    navigator.geolocation.clearWatch(this.watchID);
```

Comment:
Where is the Watch set on watchID now that it is no longer in componentDidMount?